### PR TITLE
`Development`: Add file upload e2e tests

### PR DESCRIPTION
**Pull request by TheZoker**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

We support programming, modeling, text, and quiz exercises in the E2E tests.

### Description

This PR adds the missing file upload exercise type to the E2E tests.

#### Code Review
- [x] Review 1
- [ ] Review 2

### CODE REVIEW

**Review by bassner — APPROVED** [review]

File: src/test/cypress/fixtures/exercise/file-upload/template.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+    "mode": "INDIVIDUAL",
+    "includedInOverallScore": "INCLUDED_COMPLETELY",
+    "numberOfAssessmentsOfCorrectionRounds": [
+        {
+            "inTime": 0,
+            "late": 0
+        }
+    ],
```
Comment (Strohgelaender) [inline src/test/cypress/fixtures/exercise/file-upload/template.json:9]: This should not be needed when creating a new exercise.
```suggestion
```

Comment (TheZoker) [reply src/test/cypress/fixtures/exercise/file-upload/template.json]: Done :)

File: src/test/cypress/e2e/exercises/file-upload/FileUploadExerciseAssessment.cy.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+const tutorFeedbackPoints = 4;
+const complaint = "That feedback wasn't very useful!";
+
+describe('File upload exercise assessment', () => {
+    let course: Course;
+    let exercise: FileUploadExercise;
+
+    before('Creates a file upload exercise and makes a student submission', () => {
+        cy.login(admin);
+        courseManagementRequest.createCourse().then((response) => {
```
Comment (pal03377) [inline src/test/cypress/e2e/exercises/file-upload/FileUploadExerciseAssessment.cy.ts:28]: Suggestion for potential future PR: rewrite using await (everywhere)

File: src/test/cypress/support/pageobjects/assessment/FileUploadExerciseAssessmentPage.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+export class FileUploadExerciseAssessmentPage extends AbstractExerciseAssessmentPage {
+    getInstructionsRootElement() {
+        return cy.get('#instructions-card');
+    }
+
+    submitFeedback() {
+        cy.get('#submit').click();
+    }
+
+    rejectComplaint(response: string, examMode: false) {
```
Comment (pal03377) [inline src/test/cypress/support/pageobjects/assessment/FileUploadExerciseAssessmentPage.ts:16]: ?

Type hint or default?

File: src/test/cypress/support/requests/CourseManagementRequests.ts [hunk 4]
```diff
@@ -420,6 +423,47 @@ export class CourseManagementRequests {
...
+            url: `${UPLOAD_EXERCISE_BASE}/${exerciseID}`,
+            method: DELETE,
+        });
+    }
+
+    makeFileUploadExerciseSubmission(exerciseId: number, file: string) {
+        return cy.request({
+            url: `${EXERCISE_BASE}${exerciseId}/file-upload-submissions`,
+            method: POST,
+            body: { submissionExerciseType: 'file-upload', file, id: null },
```
Comment (pal03377) [inline src/test/cypress/support/requests/CourseManagementRequests.ts:453]: Why is `id: null` needed here?

**Review by pal03377 — COMMENTED** [review]
Small questions:

File: src/test/cypress/e2e/exercises/file-upload/FileUploadExerciseAssessment.cy.ts (hunk 2 above)
Comment (TheZoker) [reply src/test/cypress/e2e/exercises/file-upload/FileUploadExerciseAssessment.cy.ts:28]: Good point, will look into it

File: src/test/cypress/support/requests/CourseManagementRequests.ts (hunk 4 above)
Comment (TheZoker) [reply src/test/cypress/support/requests/CourseManagementRequests.ts]: Removed

**Review by pal03377 — DISMISSED** [review]
Code looks good and tests are passing 👍

**Review by tobias-lippert — DISMISSED** [review]
Tests pass and code looks good ✔

**Pull request merged**